index.js

```diff
@@ -77,10 +77,17 @@ async function fetchImage(url) {
 }
 
 async function fetchImageId(url) {
+  const image = await fetchImage(url);
+  const id = await postImage(image);
+  return id;
+}
+
+async function fetchImageIds(urls) {
   try {
-    const image = await fetchImage(url);
-    const id = await postImage(image);
-    return id;
+    const ids = await Promise.all(
+      urls.map(async (url) => await fetchImageId(url)),
+    );
```

Comment:
Just as a tip, try to rewrite this code using Promises instead of async/await.

At the end you wold have something like `Promise.all(...).then(ids)`.

`Promise.all` requires an Array of promises to resolve: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/all

Line 88 the `async` keyword assures that you get a Promise back. The `fetchImageId` also returns a promise as it is an `async function`. Once you add `await` to its call you expect this promise to be resolved and then you return a new promise (from the `async` callback inside the `.map`).

You could simple map through `urls` and call `fetchImageId` without `async/await` needed as it already returns a promise to you.

```javascript
const promises = urls.map(fetchImageId);
const ids = Promise.all(promises);
```

All in all, the code below does the same:
```suggestion
    const ids = await Promise.all(urls.map(fetchImageId));
```

Reply:
Indeed this format is way more elegant than what I wrote! Thanks for the tip😉